### PR TITLE
feat(codes): attach error codes to the constructor

### PR DESCRIPTION
**Pull request by ajpawlicki**
## What
Adds an optional `code` argument to the constructor

## Why
So we can add error codes to the error classes we're creating in lob-api

## Related PRs
lob/hapi-format-error#13

## Next steps
* Cut a new minor version of this pkg
* Update pkg in lob/lob-api passing in new `code` parameter

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,27 +7,31 @@ const Boom = require('@hapi/boom');
...
  */
-function createBoomError(name, statusCode, message) {
+function createBoomError(name, statusCode, message, code) {
   var exports = this;
 
   function ErrorCtor () {
     this.name = name;
+    if (code) {
+      this.code = code;
+    }
```
Comment (ajpawlicki) [inline index.js:20]: Only adding the `code` if it's provided in the arguments. I dislike adding `undefined` properties to the constructor.

File: test/index.test.js [hunk 2]
```diff
@@ -47,6 +49,24 @@ describe('createBoomError', function () {
...
+  it('should create a boom error with an optional error code', () => {
+    const code = 'invalid';
+    const CodeError = createBoomError('CodeError', 422, null, code);
+    const err = new CodeError();
+    expect(err instanceof CodeError).to.be.true;
+    expect(err instanceof Error).to.be.true;
+    expect(err.code).to.eql(code);
     expect(err.output).to.eql({
       statusCode: 422,
       payload: {
```
Comment (tlhunter) [inline test/index.test.js:72]: Should the `code` property also be attached to the output payload?

**Comment by tlhunter** [comment]
This should be a SemVer Minor release once merged

File: test/index.test.js (hunk 2 above)
Comment (ajpawlicki) [reply test/index.test.js:72]: I was thinking I could just manually add the error code to the output payload in the [format error plugin](https://github.com/lob/hapi-format-error/blob/master/lib/index.js#L102-L107). Any thoughts on that?

Comment (ajpawlicki) [reply test/index.test.js:72]: Something like this:
```node
if (options.permeateErrorName && err.code) {
  err.output.payload.error.code = err.code;
}
```

Comment (ajpawlicki) [reply test/index.test.js:72]: The only reason I would prefer to do it in the format error plugin is because we "rewrite" the payload anyway there

**Review by tlhunter — APPROVED** [review]
LGTM

File: package.json [hunk 3]
```diff
@@ -7,7 +7,10 @@
     "test": "test"
   },
   "scripts": {
-    "test": "mocha test test/index.test.js"
+    "test": "mocha test test/index.test.js",
+    "release:patch": "changelog -p && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version patch && git push origin && git push origin --tags",
+    "release:minor": "changelog -m && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version minor && git push origin && git push origin --tags",
+    "release:major": "changelog -M && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version major && git push origin && git push origin --tags"
```
Comment (ajpawlicki) [inline package.json:13]: FYI adding these version release scripts

**Comment by ajpawlicki** [comment]
Alright published the new version 0.3.0

**Pull request merged**